lib/contentful.js

```diff
@@ -22,23 +22,84 @@ try {
 }
 
 /**
- * Returns the fields defined on the Broadcast content type for a given broadcastId.
+ * Prefixes Contentful identifier to given error.message.
  */
-module.exports.fetchBroadcast = function (broadcastId) {
+function contentfulError(error) {
+  const scope = error;
+  scope.message = `Contentful: ${error.message}`;
+
+  return scope;
+}
+
+/**
+ * Returns first result of a Contentful getEntries call for the given query.
+ */
+function fetchSingleEntry(query) {
   return new Promise((resolve, reject) => {
-    logger.debug(`fetchBroadcast:${broadcastId}`);
-    return client.getEntries({
-      content_type: 'broadcast',
-      'fields.broadcastId': broadcastId,
-    })
+    logger.debug(`contentful.fetchSingleEntry:${JSON.stringify(query)}`);
+    return client.getEntries(query)
     .then((entries) => {
       if (entries.items.length === 0) {
         return resolve(false);
       }
-      const broadcast = entries.items[0];
+      const entry = entries.items[0];
 
-      return resolve(broadcast.fields);
+      return resolve(entry.fields);
     })
-    .catch(error => reject(error));
+    .catch(error => reject(contentfulError(error)));
+  });
+}
+
+module.exports.fetchBroadcast = function (broadcastId) {
+  logger.debug(`contentful.fetchBroadcast:${broadcastId}`);
+
+  return fetchSingleEntry({
+    content_type: 'broadcast',
+    'fields.broadcastId': broadcastId,
+  });
+};
+
+module.exports.fetchKeyword = function (keyword) {
+  logger.debug(`contentful.fetchKeyword:${keyword}`);
+
+  return fetchSingleEntry({
+    content_type: 'keyword',
+    'fields.keyword': keyword,
   });
 };
+
+module.exports.fetchKeywordsForCampaignId = function (campaignId) {
+  logger.debug(`contentful.fetchKeywordsForCampaignId:${campaignId}`);
+  const query = {
+    content_type: 'keyword',
+    'fields.campaignId': campaignId,
+  };
+
+  return client.getEntries(query)
+    .then((response) => {
+      // We store keywords for both Thor and Production in our Keywords content type.
+      // Filter the results based on our Node environment:
+      const filteredItems = response.items.filter((item) => {
+        if (!item.fields) {
+          return false;
+        }
+        if (item.fields.environment !== process.env.NODE_ENV) {
+          logger.debug(`filtering ${process.env.NODE_ENV} keyword:${item.fields.keyword}`);
+          return false;
+        }
+        return true;
+      });
+      // Remove unnecessary fields.
+      const keywords = filteredItems.map((item) => {
+        const entry = item.fields;
+        delete entry.campaignId;
```

Comment:
Delete is not recommended, see https://github.com/airbnb/javascript#objects--rest-spread:

> Prefer the object spread operator over Object.assign to shallow-copy objects. Use the object rest operator to get a new object with certain properties omitted